syncs/syncs.go

```diff
@@ -304,3 +304,39 @@ func (wg *WaitGroup) Go(f func()) {
 		f()
 	}()
 }
+
+// TODO(https://go.dev/issue/63941): LockFunc, LockValue, and LockValues
+// are helper functions proposed upstream. Their naming and signature
+// are based on the existing OnceFunc, OnceValue, and OnceValues
+// helper functions already in the [sync] package.
+
+// LockFunc runs f while holding the lock.
+func LockFunc(lock sync.Locker, f func()) {
+	lock.Lock()
+	defer lock.Unlock()
+	f()
+}
+
+// LockValue runs f while holding the lock and returns the argument.
+func LockValue[T any](lock sync.Locker, f func() T) T {
```

Comment:
`LockValue` is kinda weird, maybe `GetWithLock` and `DoWithLock` or just `Get` and `Do` like the `must` package?

Reply:
or maybe a `locks.Get`/`locks.Do`?